lib/OpenQA/Schema/ResultSet/JobSettings.pm

```diff
@@ -47,11 +47,22 @@ sub query_for_settings {
                 }
                 push(@joins, 'siblings');
             }
-            push(
-                @conds,
-                {
-                    "$tname.key"   => $setting,
-                    "$tname.value" => $args->{$setting}});
+            if ($args->{$setting} =~ /^:\w+:/) {
+                my $worker_class = $&;
+                push(
+                    @conds,
+                    {
+                        "$tname.key"   => $setting,
+                        "$tname.value" => {'like', "$worker_class%"},
+                    });
+            }
+            else {
+                push(
+                    @conds,
+                    {
+                        "$tname.key"   => $setting,
+                        "$tname.value" => $args->{$setting}});
+            }
```

Comment:
would `"$tname.value" => $args->{$setting} =~ /^:\w+:/ ? {like => "$&%"} : $args->{$setting},` work (for you) instead of that big if/else with lots of indendation?

Reply:
Even a small if would be prefeed over this.

Reply:
this feedback is 4 days old and requires 5 minutes of work. The PR is from July! And there are still no tests for it.